File: selection_sort/selection_sort.py
#Asilbek Asqarov
#Mohirdev Maʻlumotlar Tuzilmasi va Algoritmlar darslari
#dars-04
#Selection sort

def selection_sort(arr):
    n = len(arr)

    for i in range(n):
        max_index = i

        for j in range(i + 1, n):
            if arr[max_index] < arr[j]:
                max_index = j
        arr[max_index] , arr[i] = arr[i] , arr[max_index]
    
    return arr

num_list = [12, 45, 34, 57, 89, 26, 92]
print(selection_sort(num_list))

File: selection_sort/selection_sort.cpp
// Asilbek Asqarov
// Mohirdev Maʻlumotlar Tuzilmasi va Algoritmlar darslari
// dars - 04
// Selection sort

#include <iostream>
using namespace std;

void selection_sort(int arr[], int n)
{

    for (int i = 0; i < n - 1; i++)
    {
        int min_index = i;
        for (int j = i + 1; j < n; j++)
        {
            if (arr[min_index] > arr[j])
                min_index = j;
        }
        int temp = arr[min_index];
        arr[min_index] = arr[i];
        arr[i] = temp;
    }
}

int main()
{
    int num_list[] = {12, 45, 34, 57, 89, 26, 92};
    int n = sizeof(num_list) / sizeof(num_list[0]);

    cout << "List: ";
    for (int i = 0; i < n; i++)
    {
        cout << num_list[i] << " ";
    }

    selection_sort(num_list, n);

    cout << "Sorted List: ";
    for (int i = 0; i < n; i++)
    {
        cout << num_list[i] << " ";
    }

    return 0;
}